src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+    private int index;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (index == elementData.length) {
+            resize();
+        }
+        elementData[index] = value;
+        index++;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
let's check index here as well
but the logic here will be slightly different from that in your `checkIndex` method
index check for `add()` should allow `index` to be equal to `size`